### PR TITLE
Nessus new

### PR DESCRIPTION
**Pull request by AleksaZatezalo**
Working on auxilary scanner docs. Issue https://github.com/rapid7/metasploit-framework/issues/12389. Added documentation for nessus.

# Verification 
 - [x] Start msfconsole
 - [x] run msf> use auxiliary/scanner/nessus/nessus_http_login
 - [x] run msf> info -d
 - [x] Verify that you see proper documentation

### CODE REVIEW

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+msf6 auxiliary(scanner/nessus/nessus_rest_login) > set rhosts 127.0.0.1
+rhosts => 127.0.0.1
+msf6 auxiliary(scanner/nessus/nessus_rest_login) > set password N0tpassword!
+password => N0tpassword!
+msf6 auxiliary(scanner/nessus/nessus_rest_login) > set username notuser
+username => notuser 
+msf6 auxiliary(scanner/nessus/nessus_rest_login) > run
+
+[*] Attempting to login to /stop using password list
+[+] 127.0.0.1:8834         -    Success: 'notuser:N0tpassword'!
```
Comment (jheysel-r7) [inline documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md:81]: Did you test this on a real target? If so, would you be able to add `## Installation Steps` above so future users could easily test this as well? Thanks!

**Comment by sempervictus** [comment]
Been pondering this interface for some time and I think we can do more with it.
Once authenticated, the api should let us upload custom checks to include cmd payloads (or cmd wrappers for binary ones) such that everything it scans becomes a session host for us.

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+## Vulnerable Application
+
+This module will attempt to authenticate to a Nessus server RPC interface.
+
+## Verification Steps
+1. Start msfconsole
+2. Do: `use auxiliary/scanner/nessus/nessus_rest_login`
```
Comment (adfoster-r7) [inline documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md:7]: Is this meant to be `auxiliary/scanner/nessus/nessus_http_login`?

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### VHOST
+HTTP server virtual host
+
+## Scenarios
+Specific demo of using the module that might be useful in a real world scenario.
+
+```
+msf > use scanner/nessus/nessus_rest_login
+msf6 auxiliary(scanner/nessus/nessus_rest_login) > set rhosts 127.0.0.1
```
Comment (adfoster-r7) [inline documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md:72]: The file name doesn't seem to match the module that's being documented:

```
auxiliary/scanner/nessus/nessus_http_login
```

versus:

```
documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md
```

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md (hunk 2 above)
Comment (AleksaZatezalo) [reply documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md:7]: yes. changed.

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md (hunk 3 above)
Comment (AleksaZatezalo) [reply documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md:72]: Hey, I changed it. Thank you!

File: documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md (hunk 1 above)
Comment (AleksaZatezalo) [reply documentation/modules/auxiliary/scanner/nessus/nessus_http_login.md]: This is done. Thank you for your patience.

**Comment by smcintyre-r7** [comment]
Landed after some minor tweaks to the layout and I fixed a couple of typos I saw. Thanks for your contribution.

**Comment by smcintyre-r7** [comment]
# Release Notes
This adds documentation for the `auxiliary/scanner/nessus/nessus_rest_login` module.

**Pull request merged**